libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.h

```diff
@@ -31,13 +31,12 @@
 
 namespace BearSSL {
 
-class WiFiClientSecure : public WiFiClient {
+class WiFiClientSecureCtx : public WiFiClient {
```

Comment:
I think the _client context is opaque, should the secure client context be opaque too?

Reply:
ClientContext is indeed opaque and WiFiClient is a wrapper to the instance.
WiFiClientSecure is (now) also a wrapper to WiFiClientSecureCtx.
Making the latter opaque will need to move method bodies to the `.cpp`.
This is doable, functions are virtual but I still wonder if the compiler is able to do some optimization when code is in the declaration (?).

Reply:
Yes, good point about the compiler.  The same moving from .cpp->.h saved IROM with the WString optimization, so I'd expect the same here (w/o the code in the .h, g++ needs to generate a real function call and can't optimize out even trivial things since it doesn't know about it.).

I'm fine w/it being non-opaque as it may actually end up after compiling to not be a jump-to-a-jump-to-code and just a jump-to-code.